File: Bars.Gkh/libs/B4/model/RealityObj/RealityObjectAntenna.js
Ext.define('B4.model.realityobj.RealityObjectAntenna', {
    extend: 'B4.base.Model',
    idProperty: 'Id',
    requires: [
        'B4.enums.AntennaRange',
        'B4.enums.AntennaReason',
        'B4.enums.YesNoMinus',
        'B4.enums.YesNoNotSet'
    ],
    proxy: {
        type: 'b4proxy',
        controllerName: 'RealityObjectAntenna'
    },
    fields: [
        { name: 'Id', useNull: true },
        { name: 'RealityObject' },
        { name: 'Availability', defaultValue: 30 },
        { name: 'Workability', defaultValue: 0 },
        { name: 'Range', defaultValue: 0 },
        { name: 'FileInfo' },
        { name: 'FrequencyFrom', useNull: true },
        { name: 'FrequencyTo', useNull: true },
        { name: 'NumberApartments' },
        { name: 'Reason', defaultValue: 0 }
    ]
});

File: Bars.Gkh/Entities/RealityObj/RealityObjectAntenna.cs
namespace Bars.Gkh.Entities
{
    using System;
    using Bars.B4.Modules.FileStorage;
    using Bars.Gkh.Enums;

    /// <summary>
    /// Система коллективного приема телевидения (СКПТ)
    /// </summary>
    public class RealityObjectAntenna : BaseImportableEntity
    {
        /// <summary>
        /// Жилой дом
        /// </summary>
        public virtual RealityObject RealityObject { get; set; }

        /// <summary>
        /// Наличие в составе общедомового имущества СКПТ
        /// </summary>
        public virtual YesNoNotSet Availability { get; set; }

        /// <summary>
        /// Работоспособность СКПТ
        /// </summary>
        public virtual YesNoMinus Workability { get; set; }

        /// <summary>
        /// Диапазон
        /// </summary>
        public virtual AntennaRange Range { get; set; }

        /// <summary>
        /// Частота от
        /// </summary>
        public virtual decimal FrequencyFrom { get; set; }

        /// <summary>
        /// Частота до
        /// </summary>
        public virtual decimal FrequencyTo { get; set; }

        /// <summary>
        /// Количество квартир, подключенных к СКПТ
        /// </summary>
        public virtual int NumberApartments { get; set; }

        /// <summary>
        /// Файл
        /// </summary>
        public virtual FileInfo FileInfo { get; set; }

        /// <summary>
        /// Причины отсутствия СКПТ
        /// </summary>
        public virtual AntennaReason Reason { get; set; }
    }
}